.changeset/thick-cats-compete.md

```diff
@@ -0,0 +1,55 @@
+---
+'astro': minor
+---
+
+The `experimental.rewriting` feature introduced behind a flag in [v4.8.0](https://github.com/withastro/astro/blob/main/packages/astro/CHANGELOG.md#480) is no longer experimental and is available for general use.
+
+`Astro.rewrite()`/`ctw.rewrite()` allows to **render
+** another page without changing the URL of the browser in Astro pages and endpoints.
+
+```astro
+---
+// src/pages/dashboard.astro
+if (!Astro.props.allowed) {
+  return Astro.rewrite('/');
+}
+---
+```
+
```

Comment:
```suggestion

Use `context.rewrite()` in endpoints, for example to ....

```

Can we have a sentence here that introduces the code sample (also OK to update the code sample to something that makes a better example to describe!) Similar to how I added an example sentence above?